sos/plugins/apache.py

```diff
@@ -25,6 +25,10 @@ def setup(self):
         # collect list of installed modules
         self.add_cmd_output(["apachectl -M"])
 
+        # determine how much logs to collect
+        self.limit = (None if self.get_option("all_logs")
+                      else self.get_option("log_size"))
```

Comment:
Can't we just remove `self.limit` entirely from the plugin?

`add_copy_spec()` automatically checks for `--all-logs` and has the implicit `--log-size` otherwise:

```python
    def add_copy_spec(self, copyspecs, sizelimit=None, tailit=True):
        """Add a file or glob but limit it to sizelimit megabytes. If fname is
        a single file the file will be tailed to meet sizelimit. If the first
        file in a glob is too large it will be tailed to meet the sizelimit.
        """
        if sizelimit is None:
            sizelimit = self.get_option("log_size")

        if self.get_option('all_logs'):
            sizelimit = None
```